src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,130 @@
+package duke;
+
+import java.util.ArrayList;
+import java.io.Serializable;
+
+/**
+ * Tasklist for duke. 
+ * 
+ * @author Tianqi-Zhu
+ */
+public class TaskList implements Serializable {
+    private ArrayList<Task> taskList; 
+
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList; 
+    }
+
+    public TaskList() {
+        taskList = new ArrayList<Task>();
+    }
+
+    public ArrayList<Task> tasks() {
+        return taskList;
+    }
+
+    public int taskAmount() {
+        return taskList.size();
+    }
+
+    /**
+     * Add a task to the end of the list. Print the task name and total task amount in the list after adding. For Gui.
+     * 
+     * @param task Task to add to the list. 
+     */
+    public String addTask(Task task) {
+        taskList.add(task);
+        String out = "Got it. I've added this task:\n  " + Ui.SPACE_STRING + task + "\n" + Ui.SPACE_STRING +
+                "Now you have " + taskList.size() + " tasks in the list.";
+        Storage.save(this);
+        return out;
+    }
+    
+    /**
+     * Delete a task by its index. Print the task name and total task amount in the list after adding. For Gui.
+     * 
+     * @param index Index of the task to be deleted. 
+     */
+    public String deteteTask(int index) {
```

Comment:
Perhaps the name can be changed to deleteTask? (Another very minor change that can enhance readability).